go/obscuronode/host/p2p/p2p.go

```diff
@@ -108,16 +109,13 @@ func (p *p2pImpl) handleConnections(callback host.P2PCallback) {
 // Receives and decodes a P2P message, and pushes it to the correct channel.
 func (p *p2pImpl) handle(conn net.Conn, callback host.P2PCallback) {
 	if conn != nil {
-		defer func(conn net.Conn) {
-			if closeErr := conn.Close(); closeErr != nil {
-				panic(closeErr)
-			}
-		}(conn)
+		defer conn.Close()
 	}
 
 	encodedMsg, err := ioutil.ReadAll(conn)
 	if err != nil {
-		panic(err)
+		nodecommon.LogWithID(p.nodeID, "failed to read message from peer: %v", err)
```

Comment:
do we want to do this ?

Reply:
I think so. It fits with general behaviour here, which is not to freak if we get a bad connection.

Reply:
I wouldn't want to change behaviours in such a big refactor, but we can zero-in on this at a later stage